receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/docker_stats.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsecscontainermetrics
+
+// ContainerStats defines the structure for container stats
+type ContainerStats struct {
```

Comment:
Is this and the other structs for parsing the response from ECS Task Metadata?

If so... would it be a bad idea to import the structs from the ECS Agent? The ECS Agent is in Go and uses go code to generate the output that you are processing.

I guess importing that code might not be wise... ECS Agent isn't meant to be used as a library. Still you could add a link to those structures as a reference. Or a link to the ECS Task Metadata docs explaining where all of this comes from.

I think its here and here:
- https://github.com/aws/amazon-ecs-agent/blob/master/agent/handlers/v4/stats_response.go
- https://github.com/aws/amazon-ecs-agent/blob/master/agent/containermetadata/types.go

Looks like they actually mostly just import from Docker though... would it be a good idea to import the docker structs?

Reply:
Thanks for noticing this! Agree to not import from a package not meant for consumption as a library, the ECS agent. But do definitely reuse docker structs where you can that's nice.

Reply:
Thats a good idea. However, our docker-stats json structure is little bit different. It has an extra item (block) for network rate metrics which is added by ECS agent. In future, we will have another block of data for storage read/write metrics. So, creating a simple struct with the fields we care about only might be a straight-forward solution. 
```
"networkspersec": {
                "eth0": [
                    {
                        "timestamp": "2020-06-17T21:45:26.381091349Z",
                        "rx_bytes_per_sec": 20,
                        "tx_bytes_per_sec": 25
                    },
                    {
                        "timestamp": "2020-06-17T21:45:26.481091349Z",
                        "rx_bytes_per_sec": 10,
                        "tx_bytes_per_sec": 55
                    }

                ]
          }
```

Reply:
Just to confirm @hossain-rayhan you don't intend to import from the docker api types?  I see most of these are pointers, which docker doesn't use, so I understand coercing them to match your usage may be undesired in the long run.

Also, if these are largely taken from the ECS agent or related lib, some attribution would be helpful/appreciated.